src/Data/Fin/Induction.agda

```diff
@@ -59,6 +59,20 @@ open WF public using (Acc; acc)
   induct {suc i} (acc rec) = Pᵢ⇒Pᵢ₊₁ i (induct (rec (inject₁ i) i<i+1))
     where i<i+1 = ℕ<⇒inject₁< (i<1+i i)
 
+≤-Induction : {k : Fin (suc n)} (P : ∀ {p} → k ≤ p → Set ℓ)
+  (pk : P ≤-refl)
+  (ps : ∀ {m k≤m} → P {inject₁ m} k≤m → P (≤-inj→suc k≤m))
+  → ∀ {m} k≤m → P {m} k≤m
+≤-Induction {k = k} P pk ps = induct (<-wellFounded _) (compare' k _)
+  where
+  induct : ∀ {m} → Acc _<_ m → Ordering' k m → ∀ k≤m → P {m} k≤m
+  induct {suc m} (acc rs) (less (s≤s k≤m)) _ = subst P (≤-irrelevant _ _) (ps finB)
```

Comment:
Since `≤-irrelevant` does hold, there should be an analog to `cast` (for `Fin`) that computes, unlike `subst`. I was able to make various proofs `subst`-free because of that.

Reply:
What is your idea to remove `subst`?
I thought about changing `P {m} k≤m` to `Σ[ k P {m} k≤m` to `Σ[ k≤m₂ ∈ k ≤ m ] (P m k≤m₂)`.

Reply:
I was starting to realize that my explanation of how to replace `subst` here was 90% of the work to actually do it, so maybe I should do that instead.